scripts/metrics/pytest_metrics_xml_to_csv.py

```diff
@@ -0,0 +1,166 @@
+    )
+    parser.add_argument(
+        "-b", "--branch", action="store", default="",
+        help="The branch the metrics were generated on. Used for directory saving."
+    )
+    parser.add_argument(
+        "-w", "--weeks", action="store_true",
+        help="Boolean whether to include the week number in the created csv path."
+    )
+    args = parser.parse_args()
+    return args
+
+
+def _determine_test_result(test_case_node):
+    # type (xml.etree.ElementTree.Element) -> str
+    """
+    Inspects the test case node and determines the test result based on the presence of known element
+    names such as 'error', 'failure' and 'skipped'. If the elements are not found, the test case is assumed
+    to have passed. This is how the JUnit XML schema is generated for failed tests.
+
+    :param test_case_node: The element node for the test case.
+    :return: The test result
+    """
+    # Mapping from JUnit elements to test result to keep it consistent with CTest result reporting.
+    failure_elements = [('error', 'failed'), ('failure', 'failed'), ('skipped', 'skipped')]
+
+    for element in failure_elements:
+        if test_case_node.find(element[0]) is not None:
+            return element[1]
+    return 'passed'
+
+
+def parse_pytest_xmls_to_csv(full_xml_path, writer):
+    # type (str, DictWriter) -> None
+    """
+    Parses the PyTest xml file to write to a csv file. The structure of the PyTest xml is assumed to be as followed:
+    <testcase classname="ExampleClass" file="SamplePath\test_Sample.py" line="43" name="test_Sample" time="113.225">
+        <properties>
+            <property name="test_case_id" value="IDVal"/>
+        </properties>
+        <system-out></system-out>
+    </testcase>
+    :param full_xml_path: The full path to the xml file
+    :param writer: The DictWriter object to write to the csv file.
+    :return: None
+    """
+    xml_root = xmlElementTree.parse(full_xml_path).getroot()
+    test_data_dict = {}
+    # Each PyTest test module will have a Test entry
+    try:
+        for test in xml_root.findall('./testsuite/testcase'):
+            # There are some testcase fields that are not tests. They do not have 'name' attributes are skipped
+            if 'name' not in test.attrib:
+                logger.debug(f'We found a testcase where it did not have a test field. Printing attribs:\n'
+                             f'{test.attrib}')
+                continue
+            test_data_dict['test_name'] = test.attrib['name']
+            test_data_dict['duration_seconds'] = float(test.attrib['time'])
+            # using 'status' to keep it consistent with CTest xml schema
+            test_data_dict['status'] = _determine_test_result(test)
+            # replace slashes to match codeowners file
+            test_file_path = test.attrib['file'].replace("\\", "/")
+            try:
+                sig_owner = SIG_OWNER_CACHE[test_file_path]
+            except KeyError:
```

Comment:
Instead of try/except, prefer `if test_file_path in SIG_OWNER_CACHE:`

Reply:
Fixed